### PR TITLE
Allow user-specified custom workloads 

### LINKED ISSUE #11: Run Custom Workloads
# Acceptance Criteria
- Able to run my implementation of a com.ibm...Workload through spark-bench
- Able to pass appropriate parameters to my custom workload
- Pointing spark-bench to my jar and workload class should be trivially easy and should only require two command line args.

### PR DESCRIPTION
**Pull request by showermat**
Fixes #11.

### CODE REVIEW

**Comment by showermat** [comment]
I've moved the items from `Defaults.scala` into the companion objects of their workloads, even the values that correspond to data generation rather than workloads.  I'd like to have a discussion soon about more closely coupling data generation and workloads -- having one Scala file for each datagen/workload, containing a defaults object and one or both of a data generation class and a workload class.  The object would be responsible for instantiating both of the classes, if they exist.  This might be a stupid, silly idea; it's just my first thought about the best way forward.

**Comment by ecurtin** [comment]
We should get Brad to user test this feature since he's working on building a custom workload for his stuff

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ConfigCreator.scala [hunk 1]
```diff
@@ -5,27 +5,35 @@ import com.ibm.sparktc.sparkbench.utils.SparkBenchException
...
+  private val workloads: Map[String, WorkloadDefaults] = Set(
+    PartitionAndSleepWorkload,
+    KMeansWorkload,
+    LogisticRegressionWorkload,
+    CacheTest,
+    SQLWorkload,
+    Sleep,
+    SparkPi
+  ).map(wk => wk.name -> wk).toMap
+  private def loadCustom(name: String): Option[WorkloadDefaults] = {
```
Comment (brad-kaiser) [inline cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ConfigCreator.scala]: probably want a blank line here

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ConfigCreator.scala [hunk 2]
```diff
@@ -5,27 +5,35 @@ import com.ibm.sparktc.sparkbench.utils.SparkBenchException
...
+    import scala.reflect.runtime.universe.runtimeMirror
+    val mirror = runtimeMirror(scala.reflect.runtime.universe.getClass.getClassLoader)
+    val module = mirror.staticModule(name)
+    val cls = mirror.reflectModule(module).instance.asInstanceOf[WorkloadDefaults]
+    Some(cls)
   }
-
```
Comment (brad-kaiser) [inline cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/ConfigCreator.scala]: and here

File: cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/exercise/PartitionAndSleepWorkload.scala [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
 import com.ibm.sparktc.sparkbench.utils.GeneralFunctions._
-import com.ibm.sparktc.sparkbench.utils.TimedSleepDefaults
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.types.{LongType, StringType, StructField, StructType}
 
-case class PartitionAndSleepWorkload(name: String,
-                                     input: Option[String] = None,
+object PartitionAndSleepWorkload extends WorkloadDefaults {
+  val name = "timedsleep"
+  val PARTITIONS: Int = 48
```
Comment (brad-kaiser) [inline cli/src/main/scala/com/ibm/sparktc/sparkbench/workload/exercise/PartitionAndSleepWorkload.scala]: We should probably not have these be in all caps for no reason

**Pull request merged**